src/intrprtr.c

```diff
@@ -924,19 +896,18 @@ void            IntrWhileEnd ( void )
     /* code a function expression (with one statement in the body)         */
     CodeFuncExprEnd( 1UL, 0UL );
 
-
     /* switch back to immediate mode, get the function                     */
     STATE(IntrCoding) = 0;
     CodeEnd( 0 );
 
     /* If we are in a break loop, then we will have created a "dummy" local
        variable names list to get the counts right. Remove it */
-    if (STATE(CountNams) > 0)
-      STATE(CountNams)--;
+    const UInt len = LEN_PLIST(STATE(StackNams));
+    if (len > 0)
+      SET_LEN_PLIST(STATE(StackNams), len - 1);
```

Comment:
As discussed elsewhere, good practice is to zero out the unused entries in this situation, but it should be done explicitly when it is known to be correct.